pkg/controller/servicemesh/webhook/controller.go

```diff
@@ -0,0 +1,205 @@
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &reconciler{ResourceManager: common.ResourceManager{Client: mgr.GetClient(), Log: log}, scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("webhook-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch secrets, but only where name is in secretNames
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			return contains(secretNames, event.Meta.GetName())
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			return contains(secretNames, event.MetaNew.GetName())
+		},
+		// deletion and generic events don't interest us
+		DeleteFunc: func(event event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(event event.GenericEvent) bool {
+			return false
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch MutatingWebhookConfigurations
+	err = c.Watch(&source.Kind{Type: &v1beta1.MutatingWebhookConfiguration{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(ns handler.MapObject) []reconcile.Request {
+			requests := []reconcile.Request{}
+			if strings.HasPrefix(ns.Meta.GetName(), injectorWebhookConfigNamePrefix) {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      secretNames[0],
+						Namespace: ns.Meta.GetName()[len(injectorWebhookConfigNamePrefix):],
+					},
+				})
+			}
+			return requests
+		})})
+	if err != nil {
+		return err
+	}
+
+	// Watch ValidatingWebhookConfigurations
+	err = c.Watch(&source.Kind{Type: &v1beta1.ValidatingWebhookConfiguration{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(ns handler.MapObject) []reconcile.Request {
+			requests := []reconcile.Request{}
+			if strings.HasPrefix(ns.Meta.GetName(), galleyWebhookConfigNamePrefix) {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      secretNames[1],
+						Namespace: ns.Meta.GetName()[len(galleyWebhookConfigNamePrefix):],
+					},
+				})
+			}
+			return requests
+		})})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// reconciles webhook configurations
+type reconciler struct {
+	common.ResourceManager
+	scheme *runtime.Scheme
+}
+
+// Reconcile reacts to changes in secrets by updating the corresponding webhook
+// configurations
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
```

Comment:
I know this is how all our other controllers log things, but I'd like to see `Request.Namespace` changed to `Secret.Namespace` and `Request.Name` to `Secret.Name`. 

Actually, ideally this would just be `log.WithValues("Secret", request.Namespace + "/" + request.Name)`.